### PR TITLE
Explained how Sphinx does TOCs

### PR DESCRIPTION
**Pull request by boltzmann-brain**
I found the way Sphinx does TOCs one of the more difficult things to learn about Sphinx, so I added a section on this to the documentation guide on Sphinx. Also fixed some broken links and typos in the same topic.

### CODE REVIEW

File: docs/guide/tools/sphinx.rst [hunk 1]
```diff
@@ -89,8 +123,10 @@ directory.
 When you write a document, figure out if there is already a place for it in
 the project, otherwise feel free to start a new file.
 
-.. warning:: If you make a new file, make sure it is included in the *Table
-    of Contents* in ``index.rst``.
+.. warning:: If you make a new file, make sure it is included in a
+	     ``toctree`` directive in some file which is itself in a
+	     ``toctree`` directive. Otherwise, the new file will not
+	     appear in the table of contents.
```
Comment (plaindocs) [inline docs/guide/tools/sphinx.rst:129]: Might be worth mentioning that sphinx does warn you about these? What do you think?

**Review by plaindocs — APPROVED** [review]
This is great!

File: docs/guide/tools/sphinx.rst (hunk 1 above)
Comment (boltzmann-brain) [reply docs/guide/tools/sphinx.rst:129]: That's a good idea. I'll include that edit in my next pull request.

**Comment by plaindocs** [comment]
Thanks.

**Pull request merged**